### ISSUE TITLE
What counts as an exception?

### ISSUE THREAD

**Issue by wibblymat** [WebAssembly/exception-handling#6]
Should things like `unreachable`, out-of-bounds memory access, `call_indirect` with an invalid table index, etc., become catchable exceptions?

**Comment by KarlSchimpf** [comment]
I agree that these should produce an exception. They can be caught by importing the corresponding exceptions, and then using the corresponding exception index to catch them.
